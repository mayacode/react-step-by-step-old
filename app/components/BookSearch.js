import React, { Fragment } from 'react';

const BookSearch = ({ changeHandler }) => (
  <Fragment>
    <input
      type="text"
      placeholder="Put title or author"
      onChange={changeHandler}
    />
    <button>Search!</button>
  </Fragment>
);

BookSearch.displayName = 'BookSearch';

export default BookSearch;
