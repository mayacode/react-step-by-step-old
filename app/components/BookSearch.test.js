import React from 'react';
import { shallow } from 'enzyme';

import BookSearch from './BookSearch';

describe('<BookSearch />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<BookSearch changeHandler={() => undefined} />);
    expect(wrapper).toMatchSnapshot();
  });
});