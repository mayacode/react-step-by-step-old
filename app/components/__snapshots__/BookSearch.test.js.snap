// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookSearch /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookSearch
    changeHandler={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Put title or author"
          type="text"
        />,
        <button>
          Search!
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Put title or author",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search!",
        },
        "ref": null,
        "rendered": "Search!",
        "type": "button",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Put title or author"
            type="text"
          />,
          <button>
            Search!
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Put title or author",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search!",
          },
          "ref": null,
          "rendered": "Search!",
          "type": "button",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
